pylint/lint/pylinter.py

```diff
@@ -515,6 +515,13 @@ def make_options() -> Tuple[Tuple[str, OptionDict], ...]:
                     ),
                 },
             ),
+            (
+                "discover-files",
```

Comment:
Haven't done a full review, but isn't this normally called `--recursive/-r` or is `--discover-files` doing something different compared to the `--recursive` options found in many other tools?

Reply:
I was thinking about `--recursive/-r` but I later decided to use `discover` word since the PR lints only `.py` files. **But** I don't have any preference for name. If there will be an agreement on any other name, I will gladly change it.

Reply:
I'd prefer recursive. Seems more in line with other tools.

Reply:
OK I will change that to recursive